include/seqan3/search/fm_index/fm_index_cursor.hpp

```diff
@@ -32,6 +32,45 @@ namespace seqan3
  * \{
  */
 
+//!\brief The underlying suffix array intervals.
+struct suffix_array_interval
+{
+    //!\brief The begin position of the interval ("left boundary").
+    size_t begin_position;
+    //!\brief The exclusive end position of the interval ("right boundary").
+    size_t end_position;
+
+    //!\brief Convert to tuple.
+    operator std::tuple<size_t &, size_t &>()
```

Comment:
```suggestion
    operator std::tuple<size_t &, size_t &>() &
```

Reply:
I don't know, you can also do this:

```c++

struct foo
{
  int a;
  int b;
};

int main()
{
    foo bar{1, 3};

    auto & [a, b] = bar;

    b = 5;

    return bar.b; // has value 5
}
```

so we don't need a std::tie.

Reply:
But what I want is 
```cpp
int main()
{
    foo bar{1, 3};

    auto & [a, b] = bar;

    b = 5;

    foo bar2{2, 4};
    std::tie(a, b) = bar2;

    return bar.b; // has value 5
}
```

Reply:
I tried the `&`:

https://godbolt.org/z/4E6KcT

It doesn't work because .. ehm .. not so sure...
It would work if we the suffix_array_interval struct wasn't generated by another struct.